src/service/chat.js

```diff
@@ -206,6 +206,58 @@ const moveChatToAnon = async ({
   }
 };
 
+const initChatFromPost = async ({source, id}) => {
+  let payload = {source};
+  switch (source) {
+    case 'post':
+      payload = {...payload, postId: id};
+      break;
+    case 'comment':
+      payload = {...payload, commentId: id};
+      break;
+    default:
+      break;
+  }
+
+  try {
+    const response = await api.post('/chat/init-chat-from-post', payload);
+    if (response?.status === 200) {
+      return Promise.resolve(response.data);
+    }
+
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
+};
+
+const initChatFromPostAnon = async ({source, id}) => {
+  let payload = {source};
+  switch (source) {
+    case 'post':
+      payload = {...payload, postId: id};
+      break;
+    case 'comment':
+      payload = {...payload, commentId: id};
+      break;
+    default:
+      break;
+  }
+
+  try {
+    const response = await anonymousApi.post('/chat/init-chat-from-post', payload);
+    if (response?.status === 200) {
+      return Promise.resolve(response.data);
+    }
+
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `initChatFromPostAnon` is almost identical to `initChatFromPost`, except it uses `anonymousApi` instead of `api`. This is a clear violation of the DRY (Don't Repeat Yourself) principle. Consider refactoring these two functions into one, which accepts the API instance as a parameter.

```javascript
const initChatFromPost = async ({source, id}, apiInstance) => {
  let payload = {source};
  switch (source) {
    case 'post':
      payload = {...payload, postId: id};
      break;
    case 'comment':
      payload = {...payload, commentId: id};
      break;
    default:
      throw new Error(`Unsupported source: ${source}`);
  }

  try {
    const response = await apiInstance.post('/chat/init-chat-from-post', payload);
    if (response?.status === 200) {
      return Promise.resolve(response.data);
    }

    return Promise.reject(response.data);
  } catch (e) {
    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
    return Promise.reject(e);
  }
};
```
Then you can call this function with either `api` or `anonymousApi` as needed.



<!-- This is an auto-generated comment by OSS CodeRabbit -->